lading/src/generator/file_gen/model.rs

```diff
@@ -310,32 +334,131 @@ impl State {
+                }
+
+                // Bump the Inode index
+                self.next_inode = self.next_inode.saturating_add(1);
+
+                // Now, search through the peers of this File and rotate them,
+                // keeping track of the last one which will need to be
+                // deleted. There is no previous node as the rotated_inode is of
+                // the 0th ordinal.
+                let mut current_inode = rotated_inode;
+                assert!(ordinal == 0);
+                let mut prev_inode = None;
+
+                loop {
+                    let (remove_current, next_peer) = {
+                        let node = self.nodes.get_mut(&current_inode).expect("Node must exist");
+                        match node {
+                            Node::File { file } => {
+                                file.incr_ordinal();
+
+                                let remove_current = file.ordinal() > self.max_rotations;
+                                let next_peer = file.peer;
+                                (remove_current, next_peer)
+                            }
+                            Node::Directory { .. } => panic!("Expected a File node"),
+                        }
+                    };
+
+                    if remove_current {
+                        // The only time a node is removed is when it's at the
+                        // end of the line. This means that next_peer is None
+                        // and there are no further peers to explore.
+                        assert!(next_peer.is_none());
+
+                        self.nodes.remove(&current_inode);
+                        if let Some(Node::Directory { dir, .. }) = self.nodes.get_mut(&parent_inode)
+                        {
+                            dir.children.remove(&current_inode);
+                        }
+
+                        // Update the peer of the previous file to None
+                        if let Some(prev_inode) = prev_inode {
+                            let node = self.nodes.get_mut(&prev_inode).expect("Node must exist");
+                            if let Node::File { file } = node {
+                                file.peer = None;
+                            }
+                        }
+
+                        break;
+                    }
+
+                    // Move to the next peer
+                    //
+                    // SAFETY: The next_peer is only None in the
+                    // `remove_current` branch, meaning that we only reach this
+                    // point if the next peer is Some.
+                    prev_inode = Some(current_inode);
+                    if next_peer.is_none() {
+                        // We're at the end of the rotated files but not so many
+                        // it's time to rotate off.
+                        break;
+                    }
+                    current_inode = next_peer.expect("next peer must not be none");
```

Comment:
I think you could unwrap using `let`-`else` here to avoid the `expect`, and then you may not need a safety comment?

```suggestion
                    // Move to the next peer
                    prev_inode = Some(current_inode);
                    let Some(peer) = next_peer else {
                        // We're at the end of the rotated files but not so many
                        // it's time to rotate off.
                        break;
                    };
                    current_inode = peer;
```

Reply:
Right you are. I think this does in fact happen up-stack.